src/apps/api/views/competitions.py

```diff
@@ -436,13 +437,17 @@ def get_leaderboard(self, request, pk):
         columns = [col for col in query['columns']]
         users = {}
         for submission in query['submissions']:
-            if submission['owner'] not in users.keys():
-                users.update({submission['owner']: len(users)})
+            if submission['parent']:
+                submission_key = f"{submission['owner']}{submission['parent']}"
+            else:
+                submission_key = f"{submission['owner']}{submission['id']}"
+            if submission_key not in users.keys():
+                users.update({submission_key: len(users)})
```

Comment:
For a single `key: value` update, we can use the syntax, `users[submission_key] = len(users)`

Reply:
`dict.update(dict)` is really nice for updating multiple `k:v` pairs simultaneously though